python/cudf/cudf/core/groupby/groupby.py

```diff
@@ -608,6 +608,98 @@ def _scan(self, op: str, *args, **kwargs):
 
     aggregate = agg
 
+    def _head_tail(self, n, *, take_head=True):
```

Comment:
An alternative API would be to accept a slice and use that to generate the indices directly.